src/drawing.rs

```diff
@@ -360,4 +360,49 @@ impl<'a> Pencil<'a> {
         }
         self.move_origin(-position)
     }
+
+    /// Draws one of the frames of the given `animation` based on the number of times this has
+    /// been previously called.
+    pub fn draw_animation_frame(
+        &mut self,
+        animation: &mut Animation,
+        position: Vec2,
+    ) {
+        self.draw_text(animation.access_frame(), position);
+    }
+}
+
+/// An object that stores the frames of an animation.
+pub struct Animation {
+    frames: Vec<String>,
+    rate: i32,
```

Comment:
Maybe it's ok by now as it is. But other animation systems allow to configure of how long each frame. This would imply a `Frame` concept that contains the data itself (the string) and the duration.